lib/cocoapods/executable.rb

```diff
@@ -93,6 +94,21 @@ def self.popen3(bin, command, stdout, stderr)
       end
     end
 
+    def self.reader(input, output)
+      Thread.new do
+        buf = ''
+        begin
+          loop do
+            buf << input.readpartial(4096)
+            string, separator, buf = buf.rpartition(/[\r\n]/)
```

Comment:
no, this isn't correct if the buffer has more than one line ending character in it.

`"foo\nbar\n".rpartition(/[\r\n]/)  # => ["foo\nbar", "\n", ""] `